packages/core/core/src/UncommittedAsset.js

```diff
@@ -223,7 +224,7 @@ export default class UncommittedAsset {
     this.clearAST();
   }
 
-  setStream(stream: Readable) {
+  setStream(stream: () => Readable) {
```

Comment:
Maybe this needs a new name?

Reply:
I do wonder about this API. What if we instead still accepted a stream, but immediately wrote it into the cache. And when calling `getStream` we created a new stream to read it from the cache. Seems easy for plugins to mess up and always return the same stream from the function they pass or something.

Reply:
I request you don't